mujoco_warp/viewer.py

```diff
@@ -100,14 +104,37 @@ def _main(argv: Sequence[str]) -> None:
       if _ENGINE.value == "mjc":
         mujoco.mj_step(mjm, mjd)
       else:  # mjwarp
-        # TODO(robotics-simulation): recompile when changing disable flags, etc.
         wp.copy(d.ctrl, wp.array([mjd.ctrl.astype(np.float32)]))
         wp.copy(d.act, wp.array([mjd.act.astype(np.float32)]))
         wp.copy(d.xfrc_applied, wp.array([mjd.xfrc_applied.astype(np.float32)]))
         wp.copy(d.qpos, wp.array([mjd.qpos.astype(np.float32)]))
         wp.copy(d.qvel, wp.array([mjd.qvel.astype(np.float32)]))
         wp.copy(d.time, wp.array([mjd.time], dtype=wp.float32))
 
+        # on reset, update options and potentially recompile step
+        if mjd.time == 0.0:
+          recompile = False
+
+          for field in m.opt.__dataclass_fields__:
```

Comment:
i wonder if there's something more general we can do here, something like:

```py
import pickle

...

model_hash = pickle.dumps(mjm)

...

with viewer:
  while True:
    ...
    if model_hash != pickle.dumps(mjm):
      model_hash = pickle.dumps(mjm)
      m = mjwarp.put_model(mjm)
      graph = _compile_step(m, d)
```

Reply:
great suggestion, this concept is now utilized. thanks!

no need now to reset for the option changes to take effect.